.github/workflows/emscripten.yml

```diff
@@ -0,0 +1,50 @@
+name: Build with emscripten
+
+on: push
+
+permissions:
+  contents: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install cached emscripten
+        id: cache
+        uses: actions/cache@v3
+        with:
+          path: 'emsdk'
+          key: 'emsdk'
+
+      - name: Install emscripten
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          curl -L https://github.com/emscripten-core/emsdk/archive/HEAD.tar.gz | tar xz
+          mv emsdk-* emsdk
+          ./emsdk/emsdk install latest
+          ./emsdk/emsdk activate latest
+
+      - name: Build blink
+        run: |
+          . ./emsdk/emsdk_env.sh
+          emmake make -j$(nproc)
+          curl -sSL https://raw.githubusercontent.com/gzuidhof/coi-serviceworker/master/coi-serviceworker.min.js -o o//blink/blink-web/coi-serviceworker.min.js --create-dirs
```

Comment:
Is it possible for this to run without installing a service worker? If not, then this code should be vendored in the third_party folder.

Reply:
Technically the service worker is _not_ required. However, in order to host this page on _GitHub_ Pages, this service worker has to be included to enable [cross-origin isolation](https://developer.mozilla.org/en-US/docs/Web/API/crossOriginIsolated).

This is required for pages that want to access `SharedArrayBuffer` (required for WASM threads), but GitHub Pages does not enable it by default.

Reply:
Good to know. This is fine for now. I'm going to look into providing a `make MODE=nothreads` build mode so we can avoid installing service workers when we don't need them.

Reply:
Should I edit the Makefile to copy required scripts into the output folder or is doing this from the GitHub Actions script is enough?

Reply:
It only needs to be copied into the output folder if you're doing something like minifying it. I don't think we're ready for a JavaScript build system quite yet. Simply copying the non-minified source code into the repo should be sufficient.

Reply:
Please be to make sure the license and copyright are at the top of the file, if it's not there already.

Reply:
Done.